samples/snippets/publisher.py

```diff
@@ -170,6 +170,87 @@ def delete_topic(project_id: str, topic_id: str) -> None:
     # [END pubsub_delete_topic]
 
 
+def pubsub_publish_otel_tracing(
+    topic_project_id: str, trace_project_id: str, topic_id: str
+) -> None:
+    """
+    Publish to `topic_id` in `topic_project_id` with OpenTelemetry enabled.
+    Export the OpenTelemetry traces to Google Cloud Trace in project
+    `trace_project_id`
+
+    Args:
+        topic_project_id: project ID of the topic to publish to.
+        trace_project_id: project ID to export Cloud Trace to.
+        topic_id: topic ID to publish to.
+
+    Returns:
+        None
+    """
+    # [START pubsub_publish_otel_tracing]
+
+    from opentelemetry import trace
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import (
+        BatchSpanProcessor,
+    )
+    from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+    from opentelemetry.sdk.trace.sampling import TraceIdRatioBased, ParentBased
+
+    from google.cloud.pubsub_v1 import PublisherClient
+    from google.cloud.pubsub_v1.types import PublisherOptions
+
+    # TODO(developer)
+    # topic_project_id = "your-topic-project-id"
+    # trace_project_id = "your-trace-project-id"
+    # topic_id = "your-topic-id"
+
+    # In this sample, we use a Google Cloud Trace to export the OpenTelemetry
+    # traces: https://cloud.google.com/trace/docs/setup/python-ot
+    # Choose and configure the exporter for your set up accordingly.
+
+    # Sample 1 in every 1000 traces.
+    sampler = ParentBased(root=TraceIdRatioBased(1 / 1000))
+    trace.set_tracer_provider(TracerProvider(sampler=sampler))
+
+    # Export to Google Trace.
+    cloud_trace_exporter = CloudTraceSpanExporter(
+        project_id=trace_project_id,
+    )
+    trace.get_tracer_provider().add_span_processor(
+        BatchSpanProcessor(cloud_trace_exporter)
+    )
+
+    # Set the `enable_open_telemetry_tracing` option to True when creating
+    # the publisher client. This in itself is necessary and sufficient for
+    # the library to export OpenTelemetry traces. However, where the traces
+    # must be exported to needs to be configured based on your OpenTelemetry
+    # set up. Refer: https://opentelemetry.io/docs/languages/python/exporters/
+    publisher = PublisherClient(
+        publisher_options=PublisherOptions(
+            enable_open_telemetry_tracing=True,
+        ),
+    )
+
+    # The `topic_path` method creates a fully qualified identifier
+    # in the form `projects/{project_id}/topics/{topic_id}`
+    topic_path = publisher.topic_path(topic_project_id, topic_id)
+
+    tracer = trace.get_tracer("google.cloud.pubsub_v1.publisher")
+    with tracer.start_as_current_span("parent cloud trace span"):
```

Comment:
I think we should omit this, since it's not necessary to wrap the traces in this way.

Reply:
Done. This would mean, there would be no parent trace. But as discussed offline, we are good with this / this is intended.